### PR TITLE
LPS-158180, LPS-158769 

### PR DESCRIPTION
**Pull request by marcelabc**
Related issue: https://issues.liferay.com/browse/LPS-158180
Related issue: https://issues.liferay.com/browse/LPS-158769

### CODE REVIEW

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/object_definitions/edit_object_view.jsp [hunk 1]
```diff
@@ -29,7 +29,7 @@ ObjectView objectView = (ObjectView)request.getAttribute(ObjectWebKeys.OBJECT_VI
 		).put(
 			"objectViewId", objectView.getObjectViewId()
 		).put(
-			"workflowStatusJSONArray", objectDefinitionsViewsDisplayContext.getWorkflowStatusJSONArray()
+			"workflowStatusJSONArray", LocalizedJSONArrayUtil.getWorkflowStatusJSONArray(objectDefinitionsViewsDisplayContext.getLocale())
```
Comment (gabrielwas) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/object_definitions/edit_object_view.jsp:32]: @marcelabc can you use just `locale` here?

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsFieldsDisplayContext.java [hunk 2]
```diff
@@ -129,6 +129,10 @@ public String[] getForbiddenLastCharacters() {
 		return forbiddenLastCharacters.toArray(new String[0]);
 	}
 
+	public Locale getLocale() {
+		return objectRequestHelper.getLocale();
+	}
+
```
Comment (gabrielwas) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsFieldsDisplayContext.java:135]: @marcelabc I think we don't need this here if we can use `locale` variable in the JSP.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/object_definitions/edit_object_field.jsp [hunk 3]
```diff
@@ -45,6 +45,8 @@ ObjectField objectField = (ObjectField)request.getAttribute(ObjectWebKeys.OBJECT
 			"objectName", objectDefinition.getShortName()
 		).put(
 			"readOnly", !objectDefinitionsFieldsDisplayContext.hasUpdateObjectDefinitionPermission()
+		).put(
+			"workflowStatusJSONArray", LocalizedJSONArrayUtil.getWorkflowStatusJSONArray(objectDefinitionsFieldsDisplayContext.getLocale())
```
Comment (gabrielwas) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/object_definitions/edit_object_field.jsp:49]: @marcelabc can you use just locale here?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AggregationObjectFieldBusinessType.java [hunk 4]
```diff
@@ -171,9 +184,137 @@ public void validateObjectFieldSettings(
...
 		}
+
+		try {
+			ObjectRelationship objectRelationship =
+				_objectRelationshipLocalService.getObjectRelationship(
+					objectDefinitionId,
+					GetterUtil.getString(
+						objectFieldSettingsValuesMap.get("relationship")));
+
+			ObjectDefinition relatedObjectDefinition =
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AggregationObjectFieldBusinessType.java:195]: @marcelabc can you rename to `objectDefinition`?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AggregationObjectFieldBusinessType.java [hunk 5]
```diff
@@ -171,9 +184,137 @@ public void validateObjectFieldSettings(
...
+				_objectDefinitionLocalService.fetchObjectDefinition(
+					objectRelationship.getObjectDefinitionId2());
+
+			ObjectField summarizeField =
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AggregationObjectFieldBusinessType.java:199]: @marcelabc can you rename to `objectField`?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AggregationObjectFieldBusinessType.java [hunk 6]
```diff
@@ -171,9 +184,137 @@ public void validateObjectFieldSettings(
...
+				GetterUtil.getString(
+					objectFieldSettingsValuesMap.get("summarizeField")),
+				noSuchObjectFieldException);
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(portalException);
+		}
+	}
+
+	private Object _getObjectFieldSettingValue(
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AggregationObjectFieldBusinessType.java:241]: @marcelabc can we inline this method?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AggregationObjectFieldBusinessType.java [hunk 7]
```diff
@@ -171,9 +184,137 @@ public void validateObjectFieldSettings(
...
+		if (ListUtil.isEmpty(objectFilters)) {
+			return;
+		}
+
+		for (ObjectFilter objectFilter : objectFilters) {
+			Set<String> missingObjectFilterValues = new HashSet<>();
+
+			if (Validator.isNull(objectFilter.getFilterBy())) {
+				missingObjectFilterValues.add("filterBy");
+			}
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AggregationObjectFieldBusinessType.java:265]: @marcelabc can we turn this into a method?

```
if (Validator.isNull(objectFilter.getFilterBy())) {
				missingObjectFilterValues.add("filterBy");
			}
```

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AggregationObjectFieldBusinessType.java [hunk 8]
```diff
@@ -171,9 +184,137 @@ public void validateObjectFieldSettings(
...
+					ObjectFieldConstants.BUSINESS_TYPE_AGGREGATION)) {
+
+				throw new ObjectFieldSettingValueException.InvalidValue(
+					objectFieldName, "filterBy",
+					GetterUtil.getString(objectFilter.getFilterBy()));
+			}
+		}
 	}
 
+	private static final String[] _FUNCTION_TYPES = {
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/AggregationObjectFieldBusinessType.java:297]: @marcelabc just FUNCTIONS

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java [hunk 9]
```diff
@@ -554,7 +554,10 @@ private void _validateObjectViewSortColumns(
 
 				if (Objects.equals(
 						objectField.getBusinessType(),
-						ObjectFieldConstants.BUSINESS_TYPE_RELATIONSHIP)) {
+						ObjectFieldConstants.BUSINESS_TYPE_RELATIONSHIP) ||
+					Objects.equals(
+						objectField.getBusinessType(),
+						ObjectFieldConstants.BUSINESS_TYPE_AGGREGATION)) {
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java:560]: @marcelabc can we sort this?

**Review by gabrielwas — CHANGES_REQUESTED** [review]

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by gabrielwas** [comment]
Resent here: https://github.com/liferay-objects/liferay-portal/pull/852

**Pull request closed without merging**